### PR TITLE
Change _format() by fmt::format

### LINKED ISSUE #848: replace _format with fmt::format
`fmt` has deprecated its UDL-based `_format` API. This is starting to through warnings in our build: https://github.com/BlueBrain/nmodl/runs/6081886419?check_suite_focus=true#step:12:135

We should switch to the `fmt::format` API, which also provides more compile-time checks.

### PR DESCRIPTION
**Pull request by alkino**
_format() is now deprecated with fmt

Fix #848 

TODO: rebase llvm branch

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -179,80 +179,92 @@ int main(int argc, const char* argv[]) {
     app.add_option("--scratch", scratch_dir, "Directory for intermediate code output")
         ->capture_default_str()
         ->ignore_case();
-    app.add_option("--units", units_dir, "Directory of units lib file")->capture_default_str()->ignore_case();
```
Comment (ohm314) [inline src/main.cpp:182]: this is an unrelated change, is that because you changed clang-format?

**Review by ohm314 — COMMENTED** [review]
Code builds fine and changes look reasoanble. But you should remove all mentions of `using namespace fmt::literals;` since they should not be needed anymore.

File: src/main.cpp (hunk 1 above)
Comment (alkino) [reply src/main.cpp:182]: No idea why. But it is mandatory

**Review by iomaganaris — APPROVED** [review]
LGTM 👍
